libs/math/include/math/ml/loss_functions/cross_entropy.hpp

```diff
@@ -59,17 +59,16 @@ typename ArrayType::Type CrossEntropyLoss(
   // if not a one-hot, must be binary logistic regression cost
   if (n_dims == 1)
   {
-    ASSERT(n_classes == SizeType(2));
+    assert(n_classes == SizeType(2));
 
     auto     x_it = x.cbegin();
     auto     y_it = y.cbegin();
     DataType one{1};
-    DataType zero{0};
     DataType tmp;
 
     while (x_it.is_valid())
     {
-      ASSERT((*y_it == one) || (*y_it == zero));
+      assert((*y_it == one) || (*y_it == static_cast<DataType>(0)));
```

Comment:
this variable declaration of "zero" and "one" was because it seemed needless to continually instantiate/cast a variable in a while loop. but presumably you're changed it back to a cast because the compiler can be relied upon to optimise this out?

Reply:
It leaves the variable unused in release mode as `zero` was only used here. It definitely has no performance implications for release mode and most likely not for debug mode.